### PR TITLE
3632 Migrate Opinions Search Alerts to ES

### PR DESCRIPTION
**Pull request by albertisfu**
According to #3632, this PR enables **`cl_send_alerts`** to work with ES as the search engine. It utilizes a switch **`o-es-alerts-active`** to control the timing of the transition, after users are informed of the changes and the V4 API is ready.

I remember that the setting **`WAFFLE_SWITCH_DEFAULT`** in Kubernetes is currently set to **`False`**. However, it would be worthwhile to confirm this so that we can proceed to merge this PR safely.

Confirming the changes between the current Webhook payload and the new one that works with ES.
Fields removed: **`caseNameShort, pagerank, status_exact, non_participating_judges, source`**
- **`court_exact`**: Copied from **`court_id`**.
- **`status`**: Mapped to the old values it had in Solr.
- **`type`**: Mapped to the old values it had in Solr.
- **`date_created`**: New field.

- A difference between the Solr version and the ES version is that in Solr, opinions are collapsed based on **`type asc`**. However, since in ES we changed the values of **`type`**, applying the same order to select the representative of the cluster won't return the same opinion as in Solr. Thus, in ES, the collapsing happens by sorting the opinions from the same cluster by relevance (**`score desc`**) and selecting the top one.
- Empty lists in the Solr Search API were represented by a `null` value (instead of an empty array in ES). This behavior is also emulated in the serializer to ensure compatibility.
- The **`timestamp`** field is a naive UTC value, whereas in V3, it's localized to Pacific Time. This conversion has also been applied.
- Other date fields are also cast to Pacific Time as in V3. (In V4, we can display UTC).
- The field changes were applied directly to the **`OpinionESResultSerializer`**, making it ready for use in the V3 API (I'll open a separate issue to integrate the V3 API with ES).
- Tweaked the ES alert templates to support the fields for Opinions correctly, including highlighting.
- Improved the Search Opinions Alert tests to ensure the emails and webhooks display the correct values and format.

During manual testing to verify the transition to ES works properly and to compare the content, I noticed a bug related to the cluster **`absolute_url`** not being updated in the **`OpinionDocument`** if the case name in the cluster changes. This does not affect searching since the **`absolute_url`** is taken from the **`OpinionClusterDocument`**, which is correctly updated. However, it affects the API (and webhooks) since it display of **`OpinionDocument`** instead of **`OpinionCluster`**.

Therefore, I've applied the fix to the signals. However, I'm afraid that the next `OpinionDocument` indexing, planned to start in the coming days, should include not only missing documents but also all opinions, to ensure the **`absolute_url`** is updated.

### CODE REVIEW

**Comment by albertisfu** [comment]
> I'm sorry I'm not doing a better job of this, Alberto, but shouldn't this alert tool be upgraded the way we did with oral arguments, where it's using the percolator index in real time for all alerts?

Yes, we can migrate the Opinion Alerts to use the percolator. I made the tweak to the command as a temporary solution, as we planned it in the [issue](https://github.com/freelawproject/courtlistener/issues/3632#issue-2088981235) so we can proceed to remove all Solr dependencies and turn it off once nothing else depends on Solr. As a second step, we'd move them to the Percolator. However, if it's better to do it now, I can proceed with that.

**Comment by mlissner** [comment]
Got it. Thanks for the explanation. @ERosendo, can you please review when you have a moment?

File: cl/alerts/management/commands/cl_send_alerts.py [hunk 1]
```diff
@@ -115,6 +127,9 @@ def __init__(self, *args, **kwargs):
         }
         self.options = {}
         self.valid_ids = {}
+        self.o_es_alerts = False
+        if waffle.switch_is_active("o-es-alerts-active"):
+            self.o_es_alerts = True
```
Comment (ERosendo) [inline cl/alerts/management/commands/cl_send_alerts.py:132]: I think we can use a ternary if statement:

```python
self.o_es_alerts = (
    True if waffle.switch_is_active("o-es-alerts-active") else False
)
```

**Comment by ERosendo** [comment]
LGTM. We can merge the PR once we've addressed the latest comments.

File: cl/search/api_serializers.py [hunk 2]
```diff
@@ -370,22 +376,43 @@ class OpinionESResultSerializer(DocumentSerializer):
...
+        """
+        ret = super().to_representation(instance)
+        ret["type"] = inverted_o_type_index_map.get(ret["type"])
+        ret["status"] = PRECEDENTIAL_STATUS.get_status_value_reverse(
+            ret["status"]
+        )
+        ret["court_exact"] = ret["court_id"]
+        ret["snippet"] = get_highlight(instance, "text")
+
+        return ret
```
[the remark this answers is not in the document]
Comment (albertisfu) [reply cl/search/api_serializers.py]: Great, thanks for the suggestion. I've changed the serializer to use this approach.

File: cl/alerts/tasks.py [hunk 3]
```diff
@@ -558,10 +558,10 @@ def process_percolator_response(response: PercolatorResponseType) -> None:
         alert_user: UserProfile.user = alert_triggered.user
         # Set highlight if available in response.
         if hasattr(hit.meta, "highlight"):
-            merge_highlights_into_result(
```
[the remark this answers is not in the document]
Comment (albertisfu) [reply cl/alerts/tasks.py:561]: Done!

File: cl/alerts/management/commands/cl_send_alerts.py (hunk 1 above)
Comment (albertisfu) [reply cl/alerts/management/commands/cl_send_alerts.py]: Yeah, changed!

**Comment by albertisfu** [comment]
@ERosendo Thanks, I've applied the suggestions you recommended.

File: cl/alerts/management/commands/cl_send_alerts.py (hunk 1 above)
Comment (johnludwigm) [reply cl/alerts/management/commands/cl_send_alerts.py]: or just `self.o_es_alerts = waffle.switch_is_active("o-es-alerts-active")`

**Pull request merged**